spec/factories/competitions.rb

```diff
@@ -238,7 +238,7 @@
 
     trait :with_delegate_report do
       after(:create) do |competition|
-        FactoryBot.create(:delegate_report, :posted, competition: competition)
```

Comment:
Have you tried setting this as a default in the `delegate_report` factory directly? You could introduce it as a trait, something like `with_report_url` and then reference that here